### PR TITLE
core(css-usage): ignore removed stylesheets

### PR DESCRIPTION
**Pull request by adamraine**
`CSS.getStyleSheetText` throws an error if the style sheet was removed. This causes issues in FR timespan mode because style sheets discovered during the timespan can be removed later in the timespan. Example:

```js
const puppeteer = require('puppeteer');
const fs = require('fs');
const open = require('open');
const defaultConfig = require('./lighthouse-core/fraggle-rock/config/default-config.js');
const lighthouse = require('./lighthouse-core/fraggle-rock/api.js');

async function run() {
  const browser = await puppeteer.launch({
    headless: false,
  })
  const page = await browser.newPage();
  const url = 'https://espn.com';

  const config = {
    ...defaultConfig,
    // audits: defaultConfig.audits.filter(a => a === 'byte-efficiency/unminified-css'),
    settings: {
      output: 'html'
    },
  };
  const run = await lighthouse.startTimespan({page, config});

  await page.goto(url);

  const {artifacts, report, lhr} = await run.endTimespan();
  fs.writeFileSync('lhr.json', JSON.stringify(lhr));
  fs.writeFileSync('artifacts.json', JSON.stringify(artifacts));
  fs.writeFileSync('fr-report.html', report);
  open('fr-report.html');

  await page.close();
  await browser.close();
}
run();

```

Simplest solution is to ignore any style sheet that was removed, which is implemented here. This does mean that styles are only tracked if they persist until the end of the timespan.

I experimented with a solution that ran `CSS.getStyleSheetText` as `CSS.styleSheetAdded` events rolled in rather than doing a batch run in `getArtifact`. Unfortunately, this didn't completely eliminate the risk of the style sheet being removed by the time `CSS.getStyleSheetText` is run.

### CODE REVIEW

File: lighthouse-core/test/gather/gatherers/css-usage-test.js [hunk 1]
```diff
@@ -68,14 +72,69 @@ describe('.getArtifact', () => {
...
+
   it('dedupes stylesheets', async () => {
     const driver = createMockDriver();
     driver.defaultSession.on
       .mockEvent('CSS.styleSheetAdded', {header: {styleSheetId: '1'}})
       .mockEvent('CSS.styleSheetAdded', {header: {styleSheetId: '1'}});
     driver.defaultSession.sendCommand
       .mockResponse('DOM.enable')
-      .mockResponse('CSS.enable', undefined, 1) // Force events to emit
+      // @ts-ignore - Force events to emit.
```
Comment (brendankenny) [inline lighthouse-core/test/gather/gatherers/css-usage-test.js:136]:
```suggestion
      // @ts-expect-error - Force events to emit.
```

File: lighthouse-core/test/gather/gatherers/css-usage-test.js [hunk 2]
```diff
@@ -19,7 +22,8 @@ describe('.getArtifact', () => {
       .mockEvent('CSS.styleSheetAdded', {header: {styleSheetId: '2'}});
     driver.defaultSession.sendCommand
       .mockResponse('DOM.enable')
-      .mockResponse('CSS.enable', undefined, 1) // Force events to emit
+      // @ts-ignore - Force events to emit.
```
Comment (brendankenny) [inline lighthouse-core/test/gather/gatherers/css-usage-test.js:25]:
```suggestion
      // @ts-expect-error - Force events to emit.
```

**Review by brendankenny — COMMENTED** [review]
> Simplest solution is to ignore any style sheet that was removed, which is implemented here. This does mean that styles are only tracked if they persist until the end of the timespan.

it makes sense to ignore to get it working (what else are we going to do?). Could also catch on `CSS.getStyleSheetText`, which is in the style of what we do elsewhere, but maybe explicitly handling sheet removal here is better.

What does this mean for the larger timespan experience, though? It seems like there's going to be multiple subtleties about the results you get from a few of our gatherers, and I don't feel like I have a good handle on the overall picture. For navigation currently, we just do our best and if something's not available, it's just not available. Sometimes we issue a warning, but not consistently. It seems like this might be more of a fundamental issue with snapshots and timespans, though?

**Comment by patrickhulce** [comment]
> What does this mean for the larger timespan experience, though? It seems like there's going to be multiple subtleties about the results you get from a few of our gatherers, and I don't feel like I have a good handle on the overall picture.

> It seems like this might be more of a fundamental issue with snapshots and timespans, though?

Are you referring to the fact that our traditional data isn't always available and the data we collect is very run-dependent? That *is* the fundamental issue with timespans and snapshots that all of @adamraine 's other work has been about slowly fixing 😄 We aren't getting there overnight, or even by the MVP. Immediate goal is an e2e working experience and permanent work thereafter to increase the scope of artifacts we can collect in every environment.

File: lighthouse-core/gather/gatherers/css-usage.js [hunk 3]
```diff
@@ -17,6 +17,11 @@ class CSSUsage extends FRGatherer {
     this._stylesheets = [];
     /** @param {LH.Crdp.CSS.StyleSheetAddedEvent} sheet */
     this._onStylesheetAdded = sheet => this._stylesheets.push(sheet);
+    /** @param {LH.Crdp.CSS.StyleSheetRemovedEvent} sheet */
+    this._onStylesheetRemoved = sheet => {
+      const styleSheetId = sheet.styleSheetId;
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/css-usage.js:22]:
```suggestion
      // We can't fetch the content of removed stylesheets, so we ignore them completely.
      // TODO(FR-COMPAT): Refactor use of CSSUsage artifact to not *always* rely on the content of stylesheets. 
      const styleSheetId = sheet.styleSheetId;
```

**Review by patrickhulce — APPROVED** [review]
SGTM for now! Long-term, not every audit that uses CSSUsage actually needs the content itself and the header is plenty good enough for its purposes. We can revisit breaking up this artifact at a later date, but this addresses the immediate failures quite elegantly 👍

**Pull request merged**